src/moonlink/src/storage/mooncake_table.rs

```diff
@@ -546,7 +546,7 @@ impl MooncakeTable {
         if contains_new_writes(iceberg_snapshot_res) {
             assert!(
                 self.last_iceberg_snapshot_lsn.is_none()
-                    || self.last_iceberg_snapshot_lsn.unwrap() < flush_lsn,
+                    || self.last_iceberg_snapshot_lsn.unwrap() < flush_lsn || flush_lsn == 0,
```

Comment:
I don't think this is the correct way, you could still suffer two concurrent mooncake snapshots.
one for initial copy, another for later changes

Reply:
For example, one detailed state to set is flush LSN.
If you have two mooncake snapshot ongoing, you could suffer
- snapshot A: initial copy mooncake snapshot ongoing, which indicates LSN 1
- snapshot B: you make some changes, and gets reflected to mooncake snapshot which indicates LSN 2

It's possible to have snapshot B handled before snapshot A, since both run at background threads
=> you suffer LSN regression